sql/core/src/main/scala/org/apache/spark/sql/execution/exchange/BroadcastExchangeExec.scala

```diff
@@ -105,6 +106,17 @@ case class BroadcastExchangeExec(
   @transient
   private val timeout: Long = conf.broadcastTimeout
 
+  @transient
+  private lazy val maxBroadcastRows = mode match {
+    case HashedRelationBroadcastMode(key, _)
+      if !(key.length == 1 && key.head.dataType == LongType) =>
```

Comment:
Could you add some comments about this condition, @c21 ?

Reply:
@dongjoon-hyun - sure, added.